vars/slackSendNotification.groovy

```diff
@@ -0,0 +1,39 @@
+import com.sap.piper.ConfigurationHelper
+
+import groovy.transform.Field
+import groovy.text.SimpleTemplateEngine
+
+@Field String STEP_NAME = 'slackSendNotification'
+@Field Set STEP_CONFIG_KEYS = ['baseUrl', 'channel', 'color', 'credentialsId', 'message']
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters, allowBuildFailure: true) {
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment, currentBuild: currentBuild]
+        // load default & individual configuration
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        def buildStatus = script.currentBuild.result
+        // resolve templates
+        config.color = SimpleTemplateEngine.newInstance().createTemplate(config.color).make([buildStatus: buildStatus]).toString()
```

Comment:
@CCFenner I know each statement reasolves the parameters at its own. Above I mean a `parallel` statement inside one step. When the configuration is handed over in such a parallel construct and each of of the parallel executions performs updates into the configuration we are lost since issues caused by such parallel updates are hard to troubleshoot. All that we avoid by having an immutable configuration map. In case somebody needs to do a second loop it is easy to either (1) continue with updating an already existing config helper or (2) create a new config helper with the config map we have at that point in time. In general immutability is a high value ...